SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp

```diff
@@ -84,9 +84,9 @@ void USpatialMetrics::SpatialStartRPCMetrics()
 	RPCTrackingStartTime = FPlatformTime::Seconds();
 
 	// If RPC tracking is activated on a client, send a command to the server to start tracking.
-	if (!NetDriver->IsServer())
+	if (!bIsServer && ControllerRefProvider.IsBound())
 	{
-		FUnrealObjectRef PCObjectRef = NetDriver->PackageMap->GetUnrealObjectRefFromObject(Cast<APlayerController>(NetDriver->GetSpatialOSNetConnection()->OwningActor));
+		FUnrealObjectRef PCObjectRef = ControllerRefProvider.Execute();
```

Comment:
We should probably check the validity of return `PCObjectRef`

Reply:
It's done indirectly below, checking the entity id for an invalid entity id

Reply:
Ah yes 👍